packages/asset-swapper/src/utils/market_operation_utils/comparison_price.ts

```diff
@@ -0,0 +1,79 @@
+import { Web3Wrapper } from '@0x/dev-utils';
+import { BigNumber, logUtils } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { MarketOperation } from '../../types';
+
+import { COMPARISON_PRICE_DECIMALS } from './constants';
+import { ComparisonPrice, ERC20BridgeSource, FeeEstimate, FeeSchedule, MarketSideLiquidity } from './types';
+
+/**
+ * Takes in an optimizer response and returns a price for RFQT MMs to beat
+ * returns the price of the taker asset in terms of the maker asset
+ * So the RFQT MM should aim for a higher price
+ * @param adjustedRate the adjusted rate (accounting for fees) from the optimizer, maker/taker
+ * @param amount the amount specified by the client
+ * @param marketSideLiquidity the results from querying liquidity sources
+ * @param feeSchedule the fee schedule passed to the Optimizer
+ * @return ComparisonPrice object with the prices for RFQ MMs to beat
+ */
+export function getComparisonPrices(
+    adjustedRate: BigNumber,
+    amount: BigNumber,
+    marketSideLiquidity: MarketSideLiquidity,
+    feeSchedule: FeeSchedule,
+): ComparisonPrice {
+    let wholeOrder: BigNumber | undefined;
+    let feeInEth: BigNumber | number;
+
+    // HACK: get the fee penalty of a single 0x native order
+    // The FeeSchedule function takes in a `FillData` object and returns a fee estimate in ETH
+    // We don't have fill data here, we just want the cost of a single native order, so we pass in undefined
+    // This works because the feeSchedule returns a constant for Native orders, this will need
+    // to be tweaked if the feeSchedule for native orders uses the fillData passed in
+    // 2 potential issues: there is no native fee schedule or the fee schedule depends on fill data
+    if (feeSchedule[ERC20BridgeSource.Native] === undefined) {
+        logUtils.warn('ComparisonPrice function did not find native order fee schedule');
```

Comment:
When would this happen exactly?

Reply:
this would happen if a given fee schedule does not include native orders. Since asset-swapper you to specify your own fee schedule, it is possible to not have a function defined for native orders.

Reply:
@alexkroeger is it worth having a `getFeeInETH(feeSchedule): BigNumber | undefined` function to factor out this logic here?

Reply:
This issue was actually causing the error in the test prior to fixing it.